receptorctl/noxfile.py

```diff
@@ -28,12 +29,42 @@ def install(session: nox.Session, *args, req: str, **kwargs):
     session.install("-r", requirements_directory / f"{req}.in", *args, **kwargs)
 
 
+def version(session: nox.Session):
+    """
+    Create a .VERSION file.
+    """
+    try:
+        official_version = session.run(
+            "git",
+            "describe",
+            "--exact-match",
+            "--tags",
+            external=True,
+            stderr=open(os.devnull, "w"),
```

Comment:
The `open()` call returns a file descriptor and leaves it open. So I'd expect it to leave an hanging unclosed resource in memory since you don't call `.close()` on that object, nor do you use a context manager that would do it for you.

Moreover, it's customary to pass the built-in `subprocess.DEVNULL` constant to `subprocess.run()` calls which nox's helper is supposed to use under the hood.

Reply:
Well spotted. Yeah that's my bad that I didn't call `.close()` and should have caught that.

Thinking about it now, the purpose was to suppress the `fatal:` message when `--exact-match` was not found, as it is done in the makefile. Although I did print a message that the closest annotated tag was being used instead so maybe we just keep them both so it's more informative?

Reply:
You can't call `.close()` if you don't save the file descriptor to memory...

Reply:
Hey @webknjaz could you take another look at this please? I've updated in an attempt to improve this and would appreciate some feedback. Thanks.